### PR TITLE
[b/546590839] Fix: Add Authentication URL parameter to support Wiz Gov (FedRAMP)

### PR DESCRIPTION
**Pull request by arunk794**
### Summary of Changes
- Resolved Bug **b/546590839**: Added optional `Authentication URL` configuration parameter (default: `https://auth.app.wiz.io`) to the Wiz integration.
- Enables Gov (FedRAMP) (`https://auth.app.wiz.us`), AWS GovCloud, and custom regional endpoints to authenticate properly without breaking existing commercial configurations.
- Implemented `get_auth_url` in `auth_manager.py` to sanitize base URLs and append `/oauth/token`.
- Bumped integration version to `9.0` in `pyproject.toml` and added release note entry in `release_notes.yaml`.
- Added unit tests in `test_manager.py` validating parameter resolution, trailing slash handling, and token generation.

### Verification & Testing
- All 44 unit tests passing: `44 passed in 0.20s`.
- Verified live in Google SecOps SOAR with network traces in Pettler for commercial (`200 OK`) and Gov endpoints.

Bug: b/546590839

### CODE REVIEW

File: content/response_integrations/google/wiz/core/utils.py [hunk 1]
```diff
@@ -63,6 +63,13 @@ def get_integration_parameters(chronicle_soar: ChronicleSOAR) -> IntegrationPara
             is_mandatory=True,
             print_value=True,
         )
+        auth_url: str | None = extract_job_param(
```
Comment (adarshtwy) [inline content/response_integrations/google/wiz/core/utils.py:66]: Add the parameter definition to `wiz_secops_bidirectional_sync_job.yaml`

Here, parameter extraction was added for jobs. However, `jobs/wiz_secops_bidirectional_sync_job.yaml` was not updated. In SecOps SOAR, the job parameter UI schema is generated directly from the job's YAML manifest. Because Wiz Authentication URL is missing from `wiz_secops_bidirectional_sync_job.yaml`, users running the Bi-directional Sync Job in a FedRAMP/Gov environment will not see or be able to configure this field in the UI. The job will silently fall back to the default commercial auth endpoint.

File: content/response_integrations/google/wiz/core/auth_manager.py [hunk 2]
```diff
@@ -90,6 +91,22 @@ def _authenticate_session(
...
+    """Get the full OAuth token URL from the configured Authentication URL.
+
+    Args:
+        auth_url (str | None): The Authentication URL configured in the integration.
+
+    Returns:
+        str: The full token URL ending with /oauth/token.
+
+    """
+    base_url: str = (auth_url or constants.DEFAULT_AUTH_URL).strip().rstrip("/")
```
Comment (adarshtwy) [inline content/response_integrations/google/wiz/core/auth_manager.py]: 1. In Python, a string of spaces `"   "` is truthy (`bool("   ") == True`). If a user accidentally enters whitespace in SOAR configuration, `(auth_url or constants.DEFAULT_AUTH_URL)` evaluates to `"   "`. Then `.strip().rstrip("/")` reduces it to `""` (empty string). `"".endswith("/oauth/token")` is `False`, returning `"/oauth/token"`. When `session.post("/oauth/token", ...)` is called, requests raises `requests.exceptions.MissingSchema: Invalid URL '/oauth/token': No scheme supplied.`

2. `constants.AUTH_ENDPOINT` is defined as `"/oauth/token"` in `constants.py`, but `auth_manager.py` hardcodes the literal string `"/oauth/token"` twice.

3. If a user supplies `https://auth.app.wiz.us/oauth/token/`, `rstrip("/")` trims the slash to `https://auth.app.wiz.us/oauth/token`, which is handled, but if they enter `https://auth.app.wiz.us/oauth/token/  ` with spaces, it should be stripped safely before trailing checks.

File: content/response_integrations/google/wiz/tests/test_manager.py [hunk 3]
```diff
@@ -275,3 +278,82 @@ def test_resolve_issue_success(
         assert len(script_session.request_history) == 1
         assert script_session.request_history[0].response.status_code == 200
         assert script_session.request_history[0].response.json() == common.RESOLVE_ISSUE
+
```
Comment (adarshtwy) [inline content/response_integrations/google/wiz/tests/test_manager.py:282]: 1. The test parametrization covers `None` and `""`, but lacks test cases for whitespace strings like `"   "` and URLs with trailing slashes after the path `(e.g., "https://auth.app.wiz.us/oauth/token/")`.
2. Add a test in `test_manager.py` or `test_sync_job.py` verifying that `get_integration_parameters` correctly extracts `auth_url` from both action and job contexts.

**Review by adarshtwy — CHANGES_REQUESTED** [review]

**Review by adarshtwy — APPROVED** [review]
LGTM!

**Review by KrishnaSharma06 — APPROVED** [review]
